job_bundles/tile_render_maya_ffmpeg/template.yaml

```diff
@@ -0,0 +1,331 @@
+  description: Whether to produce an error when there is an Arnold license failure.
+  default: 'false'
+  allowedValues:
+  - 'true'
+  - 'false'
+- name: OverrideAdaptorWheels
+  type: PATH
+  objectType: DIRECTORY
+  dataFlow: IN
+  userInterface:
+    control: CHOOSE_DIRECTORY
+    label: Wheels directory
+    groupLabel: Override settings
+  description: A directory that contains wheels for openjd, deadline, and the overridden
+    adaptor.
+- name: OverrideAdaptorName
+  type: STRING
+  userInterface:
+    control: LINE_EDIT
+    label: Adaptor Name
+    groupLabel: Override settings
+  description: The name of the adaptor to override, for example NukeAdaptor or MayaAdaptor.
+  default: MayaAdaptor
+steps:
+- name: masterLayer tile render
+  parameterSpace:
+    taskParameterDefinitions:
+    - name: Frame
+      type: INT
+      range: '{{Param.Frames}}'
+    - name: Camera
+      type: STRING
+      range:
+      - RenderCam
+    - name: TileNumberX
+      type: INT
+      range: '1-{{Param.NumXTiles}}'
+    - name: TileNumberY
+      type: INT
+      range: '1-{{Param.NumYTiles}}'
+  stepEnvironments:
+  - name: Maya
+    description: Runs Maya in the background.
+    script:
+      embeddedFiles:
+      - name: initData
+        filename: init-data.yaml
+        type: TEXT
+        data: |
+          renderer: arnold
+          render_layer: masterLayer
+          scene_file: '{{Param.MayaSceneFile}}'
+          project_path: '{{Param.ProjectPath}}'
+          output_file_path: '{{Param.OutputFilePath}}'
+          render_setup_include_lights: {{Param.RenderSetupIncludeLights}}
+          strict_error_checking: {{Param.StrictErrorChecking}}
+          output_file_prefix: '{{Param.OutputFilePrefix}}'
+          image_width: {{Param.ImageWidth}}
+          image_height: {{Param.ImageHeight}}
+          error_on_arnold_license_fail: {{Param.ArnoldErrorOnLicenseFailure}}
+      actions:
+        onEnter:
+          command: MayaAdaptor
+          args:
+          - daemon
+          - start
+          - --path-mapping-rules
+          - file://{{Session.PathMappingRulesFile}}
+          - --connection-file
+          - '{{Session.WorkingDirectory}}/connection.json'
+          - --init-data
+          - file://{{Env.File.initData}}
+          cancelation:
+            mode: NOTIFY_THEN_TERMINATE
+        onExit:
+          command: MayaAdaptor
+          args:
+          - daemon
+          - stop
+          - --connection-file
+          - '{{ Session.WorkingDirectory }}/connection.json'
+          cancelation:
+            mode: NOTIFY_THEN_TERMINATE
+  script:
+    embeddedFiles:
+    - name: runData
+      filename: run-data.yaml
+      type: TEXT
+      data: |
+        frame: {{Task.Param.Frame}}
+        camera: '{{Task.Param.Camera}}'
+        numXTiles: {{Param.NumXTiles}}
+        numYTiles: {{Param.NumYTiles}}
+        tileNumX: {{Task.Param.TileNumberX}}
+        tileNumY: {{Task.Param.TileNumberY}}
```

Comment:
Rendering a selected tile isn't a typical parameter that renderers provide directly in their interface, but region rendering is. Because the purpose of the adaptor is to reflect the renderer interface, it would be better to modify the adaptor to directly accept the region{Min/Max}{X/Y} values. Adding region render support to the adaptor also feels like something we should be able to merge into the adaptor and release independently of this tile render job, so you could remove everything about building the custom branch.

Then you can put the transformation from tile coordinates into region bounds into the job bundle.

Reply:
In regards to: 
_"Then you can put the transformation from tile coordinates into region bounds into the job bundle."_
How might you do that? I originally wanted to do this but struggled to come up with a way to do the calculations required for each tile's region coordinates such that each task would have access to the correct values based on the tile numbers.

Would another step need to run before the render step to calculate all the border values for each tile and store them in some sort of configuration file similar to how this is done with draft in deadline 10? If yes, how would this file be stored, and read when the task needs to specify the region coordinates?

Reply:
Another consideration; the changes I made to the adaptor includes creating the file name for the tiles, specific to the ffmpeg assembly pattern (top-down, left-right). If the logic moves to the job bundle, the filename would also need to be created there and passed into the adaptor.

Reply:
Rather than try to explain it all in words, I wrote a sample git commit that's how I would tackle these things. Note, I have not run or tested this code in any way. But I hope it's helpful!

https://github.com/mwiebe/deadline-cloud-samples/commit/ca4fa1162f504c92798536e1931b6b1c6df90285

Reply:
Thanks for adding the "next steps" in the description. Let's go ahead and merge this, and we'll be able to do a follow-up later that adjusts this.